chore(example): improve example app types (1st iteration)

guoda-puidokaite opened the pull request:
# Summary

Improved example app types (1st iteration).

<!-- Provide a short summary explaining the purpose of this pull request. -->

# Changes Made

- Converted all example app components to TS syntax and improved types.
- The errors (and therefore ignore statements) have been fixed for many files. 
- However, some files have been improved but issues still remain e.g. due to package type and other issues.
- Committing this PR as the first "type improvement" iteration, to not overwhelm with more files.
- Note: Deleted `MonorepoChecker`

# Related Issues

<!-- List any related issues or tickets, including links to them. -->

- https://github.com/cloudoperators/juno/issues/593

# Checklist

<!-- Ensure that your pull request meets the following requirements. -->

- [x] I have performed a self-review of my code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [x] I have made corresponding changes to the documentation (if applicable).
- [x] My changes generate no new warnings or errors.

barsukov commented on apps/example/src/App.tsx:
```diff
@@ -1,19 +1,18 @@
 /* eslint-disable @typescript-eslint/no-unsafe-return */
 /* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
 /* eslint-disable @typescript-eslint/no-unsafe-call */
```
Do you still wanna keep few ?

guoda-puidokaite replied:
https://github.com/cloudoperators/juno/pull/734#issuecomment-2618134334

barsukov commented on apps/example/src/components/StoreProvider.tsx:
```diff
@@ -1,44 +1,83 @@
+const StoreContext = createContext<StoreApi<StoreState> | null>(null)
+
+const StoreProvider: React.FC<StoreProviderProps> = ({ children }) => (
+  // @ts-ignore
```
stil needed?

guoda-puidokaite replied:
https://github.com/cloudoperators/juno/pull/734#issuecomment-2618134334

barsukov commented on apps/example/src/components/StoreProvider.tsx:
```diff
@@ -1,44 +1,83 @@
-/* eslint-disable @typescript-eslint/no-unsafe-argument */
-/* eslint-disable @typescript-eslint/no-unsafe-member-access */
-/* eslint-disable @typescript-eslint/no-unsafe-return */
+/* eslint-disable no-unused-vars */
```
do you need it globaly?

guoda-puidokaite replied:
Moved to line 😊

barsukov commented on apps/example/src/components/auth/HeaderUser.tsx:
```diff
@@ -1,19 +1,21 @@
-const HeaderUser = ({ login, logout }: any) => {
+interface HeaderUserProps {
+  login: () => void
+  // eslint-disable-next-line no-unused-vars
```
lets create a investigation task of this one, otherwise I would really suggest to disable this one...

guoda-puidokaite replied:
Good catch, I removed the destructuring and therefore error.

barsukov commented on apps/example/src/mocks/browser.ts:
```diff
@@ -1,23 +1,27 @@
+export const startWorker = (options: StartWorkerOptions): Promise<void> => {
+  // @ts-ignore
   worker = setupWorker(...getHandlers(options))
+  // @ts-ignore
```
maybe we could specify error here ?

guoda-puidokaite replied:
https://github.com/cloudoperators/juno/pull/734#issuecomment-2618134334

barsukov commented on apps/example/src/mocks/browser.ts:
```diff
@@ -1,23 +1,27 @@
 
-export const startWorker = (options: any) => {
+export const startWorker = (options: StartWorkerOptions): Promise<void> => {
+  // @ts-ignore
```
also this one?

guoda-puidokaite replied:
https://github.com/cloudoperators/juno/pull/734#issuecomment-2618134334

barsukov commented on apps/example/src/mocks/browser.ts:
```diff
@@ -1,23 +1,27 @@
-let worker = null
+interface StartWorkerOptions {
+  endpoint: string
+  [key: string]: any
```
is it really any?

guoda-puidokaite replied:
Ah, unknown always better, right. Habit 🤪. But not sure yet what specific types the `options` could be.

barsukov commented on apps/example/src/lib/store/index.ts:
```diff
@@ -5,13 +5,18 @@
+  createStore<StoreState>(
+    // @ts-ignore
     devtools((set, get) => ({
+      // @ts-ignore
```
why you added here ts ignore? do you want to take over it later?

guoda-puidokaite replied:
https://github.com/cloudoperators/juno/pull/734#issuecomment-2618134334

barsukov reviewed (commented):
I leave it as comment. Great job! we are really getting closer with TS here.
Still few ts-ingores was added do you plan to take them down in separate PR? if yes leave it as it is

guoda-puidokaite commented:
> I leave it as comment. Great job! we are really getting closer with TS here. Still few ts-ingores was added do you plan to take them down in separate PR? if yes leave it as it is

Thank You so much, @barsukov for the thorough review and comments. 💙 I addressed some comments.

For the rest, we had no types and many errors, so this PR is the first iteration addressing this in many files (mentioned in description).

However, it's A LOT to understand and type at once... so I decided to get in what I have so far, double check types in `ui-components` and then look at the remainder based on that, with fresh eyes. Many more file changes will be required.

That sound okay?

barsukov reviewed (approved):
Great Job
